File: LeetCode/python/791.py
class Solution(object):
    def customSortString(self, S, T):
        """
        :type S: str
        :type T: str
        :rtype: str
        """
        counter = collections.Counter(T)
        res = ""
        for s in S:
            if s in counter:
                for i in range(counter[s]):
                    res += s
                counter[s] = 0
        for c, v in counter.items():
            for i in range(v):
                res += c
        return res

File: LeetCode/cpp/791.cpp
class Solution {
public:
    string customSortString(string &S, string &T) {
        vector<int> pos(26, 26);
        for (int i = 0; i < S.size(); ++i)
            pos[S[i] - 'a'] = i;
        sort(T.begin(), T.end(), [pos](const char &t1, const char &t2) {
            return pos[t1 - 'a'] < pos[t2 - 'a'];
        });
        return T;
    }
};
